### PR TITLE
Explore page   bookmark page

### PR DESCRIPTION
**Pull request by varnita619**
added all posts functionalities
- user can create, edit and delete posts
-  add post with images or with text
- add comments, update and delete comments
- like/dislike posts
- bookmark posts and remove from bookmark

### CODE REVIEW

**Review by pritam-kr — APPROVED** [review]
- make the private route
- log in as a guest should log in with one click
- user profile edit feature is not working

File: src/Components/UpdatePostModal/UpdatePostModal.jsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+              id: post._id,
+              content: postData.content,
+              imgUrl: reader.result,
+              token: token,
+            })
+          );
+        }
+      };
+    } else {
+      console.log("sjsuidfhi")
```
Comment (manish591) [inline src/Components/UpdatePostModal/UpdatePostModal.jsx]: remove console

File: src/Pages/Bookmark/Bookmark.jsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  const { userInfo } = useSelector((state) => state.auth);
+
+  const { bookmark, posts } = useSelector((state) => state.posts);
+
+  // const bookMarkedPosts = () => {
+  //   bookmark.map((eachBookMark)=>{
+  //     return posts.find((post)=> post._id === eachBookMark)
+  //   })
+  // //  return posts.filter((post) => post._id === postId)[0];
+  // };
```
Comment (manish591) [inline src/Pages/Bookmark/Bookmark.jsx]: remove commented code if not required

File: src/store/postSlice.js [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+    }
+});
+
+export const createNewPost = createAsyncThunk('posts/createPost', async ({ content, imgUrl, token }) => {
+    try {
+        const response = await axios.post('/api/posts', { postData: { content: content, imgUrl: imgUrl, token: token } }, { headers: { authorization: token } });
+        return response.data.posts;
+    }
+    catch (error) {
+        console.log(error)
```
Comment (manish591) [inline src/store/postSlice.js:23]: use console.error for errror message

**Review by manish591 — APPROVED** [review]
The code is looking good, 
Use prettier for consistent formatting in your code.

File: src/Components/UpdatePostModal/UpdatePostModal.jsx (hunk 1 above)
Comment (varnita619) [reply src/Components/UpdatePostModal/UpdatePostModal.jsx]: done

File: src/store/postSlice.js (hunk 3 above)
Comment (varnita619) [reply src/store/postSlice.js:23]: > * make the private route
> * log in as a guest should log in with one click
> * user profile edit feature is not working

done

**Pull request merged**